### PR TITLE
feat(ui): Project Debug Files improvements

### PR DESCRIPTION
**Pull request by matejminar**
This PR improves multiple things on project debug files page (pasted from asana):

Searchbar:
It never shows the search query. While typing, it shows a loading indicator, but it doesn't search. When searching by hitting [ENTER], it reloads the entire page. Also, the bar is too narrow for the expected kind of queries (debug IDs).

Pagination:
Every pagination reloads the entire page. Instead, this could just reload the file list.

Show more information:
For most debug files, we also have the code identifier (as shown in the Images Loaded section in issue details). In the future, we might collect more information, so we need a place to show it. This may need to be a toggle, similar to issue details.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/scrollToTop.jsx [hunk 1]
```diff
@@ -5,10 +5,16 @@ import React from 'react';
...
     location: PropTypes.object,
+    disable: PropTypes.func,
   };
 
   componentDidUpdate(prevProps) {
-    if (this.props.location !== prevProps.location) {
+    const {disable, location} = this.props;
+
+    const shouldDisable =
+      typeof disable === 'function' && disable(location, prevProps.location);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/scrollToTop.jsx:15]: You might be able to use `callIfFunction()` here. ⛳

File: src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx [hunk 2]
```diff
@@ -131,37 +172,84 @@ class ProjectDebugSymbols extends AsyncView<Props, State> {
...
+
+          <Filters>
+            <Label>
+              <Checkbox
+                checked={showDetails}
+                onChange={e => {
+                  this.setState({showDetails: (e.target as HTMLInputElement).checked});
+                }}
+              />
+              {t('details')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx:186]: It isn't clear on what will happen when this is clicked. Should this be 'show details'?

File: src/sentry/static/sentry/app/views/settings/components/scrollToTop.jsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/settings/components/scrollToTop.jsx]: oh TIL, thanks!

File: src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx]: updated 👍

File: src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ class ProjectDebugSymbols extends AsyncView<Props, State> {
     return endpoints;
   }
 
-  handleDelete(id: string) {
+  handleDelete = (id: string) => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx:69]: Could we add a test for this?

Comment (matejminar) [reply src/sentry/static/sentry/app/views/settings/projectDebugFiles/index.tsx]: sure, done 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**